pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -132,24 +169,40 @@ func (c CertRotationController) SyncWorker(ctx context.Context) error {
 		return err
 	}
 
-	if _, err := c.RotatedSelfSignedCertKeySecret.EnsureTargetCertKeyPair(ctx, signingCertKeyPair, cabundleCerts); err != nil {
-		return err
+	var errs []error
+	for _, secret := range c.RotatedTargetSecrets {
+		if _, err := secret.EnsureTargetCertKeyPair(ctx, signingCertKeyPair, cabundleCerts); err != nil {
+			errs = append(errs, err)
+		}
+	}
+	if len(errs) != 0 {
+		return errors.NewAggregate(errs)
 	}
 
 	return nil
 }
 
 func (c CertRotationController) targetCertRecheckerPostRunHook(ctx context.Context, syncCtx factory.SyncContext) error {
 	// If we have a need to force rechecking the cert, use this channel to do it.
-	refresher, ok := c.RotatedSelfSignedCertKeySecret.CertCreator.(TargetCertRechecker)
-	if !ok {
-		return nil
+	var targetRefreshers []<-chan struct{}
+	for _, target := range c.RotatedTargetSecrets {
+		if refresher, ok := target.CertCreator.(TargetCertRechecker); ok {
+			targetRefreshers = append(targetRefreshers, refresher.RecheckChannel())
+		}
 	}
-	targetRefresh := refresher.RecheckChannel()
+	aggregateTargetRefresher := make(chan struct{})
+	for _, ch := range targetRefreshers {
+		go func(c <-chan struct{}) {
+			for msg := range c {
+				aggregateTargetRefresher <- msg
```

Comment:
goroutine leaking: `<-ctx.Done()`, could be a problem for integration tests that check goroutine leakages?

Reply:
`ctx.Done` would close them iiuc, but yeah, worth adding a unit test which uses runhook and ensures no goroutines are leaking